### PR TITLE
SwiGLU MLP: parameter-neutral gated activation over LeakyReLU^2

### PR DESCRIPTION
**Pull request by they-call-me-god**
## Summary

Replace `LeakyReLU(0.5)^2` with **SwiGLU** gating — the same multiplicative activation used in LLaMA, Mistral, Gemma, and PaLM.

Built on the PR #549 SOTA stack (LeakyReLU² + Legal TTT + Parallel Muon). Single change, zero parameter increase.

## The Change

```python
# Before (SOTA)
x = F.leaky_relu(F.linear(x, up_w), negative_slope=0.5).square()
out = F.linear(x, down_w)

# After (SwiGLU)
half = up_w.shape[0] // 2
gate = F.silu(F.linear(x, up_w[:half]))   # learned gating
up   = F.linear(x, up_w[half:])
out  = F.linear(gate * up, down_w)
```

## Parameter Neutrality

| Bank | Old shape | New shape |
|------|-----------|-----------|
| mlp_up_bank[i] | (1536, 512) | (2048, 512) — gate\|\|up |
| mlp_down_bank[i] | (512, 1536) | (512, 1024) |

Proof: `2 × 512 × 1536 = 3 × 512 × 1024 = 1,572,864` per layer.

## Status

Training logs pending (RunPod 8×H100). Will update with 3-seed results and final val_bpb.

## New env vars

- `USE_SWIGLU=1` (default on)
- `SWIGLU_HALF_DIM=1024`

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — SwiGLU MLP: parameter-neutral gated activation over LeakyReLU^2

**BPB:** (not parsed — see PR title) | **Compliance:** LOOKS CLEAN — score-first-per-chunk TTT (legal #1416/#1423 pattern)

**What I found in the code** (head SHA `697f4ce12cff`, file `records/track_10min_16mb/2026-03-25_SwiGLU_LeakyReLU2_LegalTTT_ParallelMuon/train_gpt.py`):

The TTT path at line **1101** implements the score-first-per-chunk pattern: each chunk is scored under `torch.no_grad()` / `inference_mode()` before the `base_model.train()` + SGD adaptation runs on that same chunk, with an `is_last_chunk` guard so the final chunk gets no adaptation pass. This is the structural shape the legal frontier uses (PRs #1416 erichroepke, #1423 aryanbhosale).

Per **Issue #402 and Issue #677**, TTT is legal when each token is scored before the adapter updates on it, and that's what the code does here — chunk `ci` is scored under weights adapted only on chunks `0..ci-1`. No `prequant_ttt_adapt_adamw(val_tokens, ...)` multi-epoch fine-tune, no scored-region SLOT, no target-in-key n-gram cache.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.04s, dim=512, layers=11, vocab=1024, code=91425 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending standard checks (3-seed validation, 16MB artifact cap, 10-min wallclock on 8×H100 SXM). The compliance picture matches the legal reference frontier and no flags were raised by the classification pass.

*Auto-classification caveat:* this review was drafted by the AST-based classifier against a template derived from manually-reviewed cluster PRs (#1420, #1450, #1487, #1541, #1529, #1533, #1518). If I've misread a subtlety in your eval path — e.g., multi-epoch TTT that I mistook for single-pass, or a target-in-key lookup I missed in a helper function — please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.04s, dim=512, layers=11, vocab=1024, code=91425 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
